JSIL/ILBlockTranslator.cs

```diff
@@ -1130,11 +1138,28 @@ bool forDynamic
             var translated = TranslateNode(node as dynamic);
 
             var statement = translated as JSStatement;
-            if (statement == null) {
-                var expression = (JSExpression)translated;
+            if (statement == null)
+            {
+                var expression = (JSExpression) translated;
 
                 if (expression != null)
+                {
+                    var ranges =
```

Comment:
This pair of selects looks like it will perform really badly, so I'd like the source map support to be behind a configuration bool.